### PR TITLE
Bump v2-contracts v3.0.1

### PR DESCRIPTION
**Pull request by aeolianeth**
## What does this PR do and why?

Closes https://github.com/jbx-protocol/juice-interface/issues/927

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the
changes._

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/shared/Project/FundingCycleDetailsCard.tsx [hunk 1]
```diff
@@ -95,7 +95,7 @@ export default function FundingCycleDetailsCard({
                       </Trans>
                     }
                   >
-                    <ExclamationCircleOutlined style={{ marginRight: 2 }} />
+                    <ExclamationCircleOutlined style={{ marginRight: 6 }} />
```
Comment (aeolianeth) [inline src/components/shared/Project/FundingCycleDetailsCard.tsx:98]: smol

File: src/components/shared/currency/ETHAmount.tsx [hunk 2]
```diff
@@ -24,18 +20,6 @@ export default function ETHAmount({
...
   padEnd?: boolean
 }) {
-  const isMaxUint =
-    BigNumber.isBigNumber(amount) &&
-    (amount.eq(constants.MaxUint256) || amount.eq(MAX_DISTRIBUTION_LIMIT))
-
-  if (isMaxUint) {
-    return (
-      <Tooltip title={t`Distribution limit infinite`}>
-        <Trans>No limit</Trans>
```
Comment (aeolianeth) [inline src/components/shared/currency/ETHAmount.tsx:34]: "No limit" doesn't apply to all uses of `ETHAmount`. Moved the check outside the component and it seemed to fix the issue reported in Discord.

File: src/components/v2/V2Create/forms/FundingForm/index.tsx [hunk 3]
```diff
@@ -278,6 +278,8 @@ export default function FundingForm({
         setTarget('0')
         break
     }
+
+    setTargetCurrency(V2_CURRENCY_ETH)
```
Comment (aeolianeth) [inline src/components/v2/V2Create/forms/FundingForm/index.tsx:282]: Reset target currency to ETH when target type is changed.

File: src/components/v2/V2Dashboard/index.tsx [hunk 4]
```diff
@@ -52,15 +50,11 @@ export default function V2Dashboard({ projectId }: { projectId: BigNumber }) {
...
-  const { data: fundingCycle, loading: fundingCycleLoading } =
+  const { data: fundingCycleResponse, loading: fundingCycleLoading } =
     useProjectCurrentFundingCycle({
       projectId,
     })
-
-  const fundingCycleMetadata = fundingCycle
-    ? decodeV2FundingCycleMetadata(fundingCycle?.metadata)
-    : undefined
+  const [fundingCycle, fundingCycleMetadata] = fundingCycleResponse ?? []
```
Comment (aeolianeth) [inline src/components/v2/V2Dashboard/index.tsx:57]: Use metadata given to us from the new contract call.

File: src/components/v2/V2Project/TreasuryStats/DistributedRatio.tsx [hunk 5]
```diff
@@ -47,11 +49,17 @@ export default function DistributedRatio({ style }: { style?: CSSProperties }) {
...
             />{' '}
-            /{' '}
-            <V2CurrencyAmount
-              amount={distributionLimit}
-              currency={distributionLimitCurrency}
-            />
+            <>
+              /{' '}
+              {distributionLimit.eq(MAX_DISTRIBUTION_LIMIT) ? (
+                <Trans>NO LIMIT</Trans>
```
Comment (aeolianeth) [inline src/components/v2/V2Project/TreasuryStats/DistributedRatio.tsx:55]: handle NO LIMIT here instead

File: src/components/v2/V2Project/V2FundingCycleSection/FundingCycleDetails.tsx [hunk 6]
```diff
@@ -42,33 +43,28 @@ import {
...
+  distributionLimitCurrency: BigNumber | undefined
 }) {
   const {
     theme: { colors },
   } = useContext(ThemeContext)
 
-  const { tokenSymbol, projectId, primaryTerminal } =
-    useContext(V2ProjectContext)
-
-  const { data: distributionLimitData } = useProjectDistributionLimit({
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2FundingCycleSection/FundingCycleDetails.tsx:55]: This caused a bug where the project preview didnt update when changing limit.

Kinda unfortunate consequence of our architecture - it's hard to know when to fetch something and when not to.

Anyway, passing in dist limit as props fixes it

File: src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx [hunk 7]
```diff
@@ -22,6 +22,94 @@ import useUsedDistributionLimit from 'hooks/v2/contractReader/UsedDistributionLi
 import { V2UserContext } from 'contexts/v2/userContext'
 import { formatDiscountRate, MAX_DISTRIBUTION_LIMIT } from 'utils/v2/math'
 
+// Fill in gaps between first funding cycle of each configuration:
+//     - derives starts from duration and start time of the first FC of that configuration
+//     - derives weights from discount rate and weight of the first FC of the configuration
+//     - derives number by incrementing
+//     - everything else the same as the first FC of the configuration
+const deriveFundingCyclesBetweenEachConfiguration = ({
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx:30]: Move this Fn outside the component as it doesn't need to be there. Helps with maintainability.

File: src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx [hunk 8]
```diff
@@ -116,104 +204,29 @@ function HistoricalFundingCycle({
...
-          interimNumber = nextInterimNumber
-          interimIndex++
-
-          allFundingCycles.push(nextFundingCycle)
-        }
-      },
-    )
-    return allFundingCycles
-  }
+  const { data: distributionLimitData } = useProjectDistributionLimit({
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx:222]: Fetch dist limit for the selected FC here

File: src/components/v2/shared/SplitItem.tsx [hunk 9]
```diff
@@ -60,7 +61,21 @@ export default function SplitItem({
...
+        <div style={{ fontWeight: 500 }}>
+          <Tooltip
+            title={<Trans>Juicebox V2 project with ID {split.projectId}</Trans>}
+          >
+            <Link
+              to={`/v2/p/${split.projectId}`}
+              target="_blank"
+              className="text-primary hover-text-action-primary hover-text-decoration-underline"
+            >
+              @{split.projectId}
```
Comment (aeolianeth) [inline src/components/v2/shared/SplitItem.tsx:74]: Make the handle a link, at least

File: src/constants/v2/ballotStrategies/index.ts [hunk 10]
```diff
@@ -6,8 +6,8 @@ import { SECONDS_IN_DAY } from 'constants/numbers'
 
 const BALLOT_ADDRESSES: { [k: string]: { [j: string]: string } } = {
   THREE_DAY: {
-    rinkeby: '0xcA65D0348E6d53BD29d52e298140375cbC6d3C0D',
-    mainnet: '0x5eaBF1D7A8c6942534cCb3489Ff221b2CBc0255b',
+    rinkeby: '0x63f683ef542779Cc151608b48aFB7Ec3E8e9BF7E',
```
Comment (aeolianeth) [inline src/constants/v2/ballotStrategies/index.ts:9]: Updated ballots

File: src/utils/v2/fundingCycle.ts [hunk 11]
```diff
@@ -65,6 +65,16 @@ const parameters: {
     parser: (val: BigNumber) =>
       val.toNumber() as V2FundingCycleMetadata['version'],
   },
+  {
```
Comment (aeolianeth) [inline src/utils/v2/fundingCycle.ts:68]: new

File: src/components/shared/currency/ETHAmount.tsx [hunk 12]
```diff
@@ -48,22 +32,21 @@ export default function ETHAmount({
...
-  if (amount) {
-    if (Number(formattedETHAmount) === 0) {
-      return (
-        <>
-          <CurrencySymbol currency="ETH" />
-          {formattedETHAmount}
-        </>
-      )
-    }
+  if (!amount) return null
```
Comment (aeolianeth) [inline src/components/shared/currency/ETHAmount.tsx:35]: Just reorging the logic for less nesting

File: src/models/v2/fundingCycle.ts [hunk 13]
```diff
@@ -7,8 +7,14 @@ export type V2FundingCycleData = {
...
 }
 
+export type V2FundingCycleMetadataGlobal = {
+  allowSetController: boolean
+  allowSetTerminals: boolean
+}
+
 export type V2FundingCycleMetadata = {
   version?: number
+  global: V2FundingCycleMetadataGlobal
```
Comment (aeolianeth) [inline src/models/v2/fundingCycle.ts:17]: Metadata updated slightly

File: src/utils/v2/fundingCycle.ts [hunk 14]
```diff
@@ -55,19 +60,38 @@ const bits1 = 0b1
...
+      return {
+        allowSetTerminals: bigNumberToBoolean(BigNumber.from(val).shr(1)),
+        allowSetController: bigNumberToBoolean(BigNumber.from(val).shr(2)),
+      }
+    },
+  },
   { name: 'reservedRate', bits: 16 },
   { name: 'redemptionRate', bits: 16, parser: invertPermyriad },
   { name: 'ballotRedemptionRate', bits: 16, parser: invertPermyriad },
+  { bits: 2 }, // 2 "spacer" bits
```
Comment (aeolianeth) [inline src/utils/v2/fundingCycle.ts:94]: There's a "gap" in the metadata of 2 bits: ballotRedemptionRate ends on bit 71, bits 72 and 73 are empty, then pausePay starts on bit 74.

Might be a bug in the contracts, not sure. Asked in Discord. Inconsequential for us anyway.

File: src/components/v2/V2Project/V2FundingCycleSection/FundingCycleDetails.tsx (hunk 6 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2FundingCycleSection/FundingCycleDetails.tsx:55]: Nice one ok

File: src/components/v2/shared/SplitItem.tsx (hunk 9 above)
Comment (johnnyd-eth) [reply src/components/v2/shared/SplitItem.tsx:74]: Nice nice

File: src/constants/v2/ballotStrategies/index.ts (hunk 10 above)
Comment (johnnyd-eth) [reply src/constants/v2/ballotStrategies/index.ts:9]: Beast

File: src/components/v2/shared/SplitItem.tsx (hunk 9 above)
Comment (johnnyd-eth) [reply src/components/v2/shared/SplitItem.tsx:74]: Nicee. Maybe @33 should <a>ID: 33</a>. ?

File: src/utils/v2/fundingCycle.ts (hunk 14 above)
Comment (johnnyd-eth) [reply src/utils/v2/fundingCycle.ts:93]: Ok cool 👍

File: src/components/shared/currency/ETHAmount.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/shared/currency/ETHAmount.tsx:34]: Ok sweet

File: src/components/v2/shared/SplitItem.tsx (hunk 9 above)
Comment (aeolianeth) [reply src/components/v2/shared/SplitItem.tsx:74]: Yeah. Idk tbh. Lets leave it for now.

**Pull request merged**